deps/config.h.template

```diff
@@ -0,0 +1,3772 @@
+/**
+ * \file config.h
```

Comment:
Maybe rename this file as `mbed-config.h.template`?

Reply:
OK. It's fine to rename it to mbedtls-config-template.h. I suggest keeping ".h" suffix to make it possible to highlight on editor or Github: much clear.